core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -16,50 +18,37 @@ sealed abstract class Signer {
 
   /**
     * The method used to sign a bitcoin unspent transaction output
-    * @param signers the [[Signer]] needed to sign the utxo
-    * @param output the utxo we are spending
-    * @param unsignedTx the unsigned transaction which is spending the utxo
-    * @param inputIndex the input index inside of the unsigned transaction which spends the utxo
-    * @param hashType the signature hashing algorithm we should use to sign the utxo
+    * @param spendingInfo - The information required for signing
+    * @param sigComponent the external TxSigComponent that needs signing
     * @param isDummySignature - do not sign the tx for real, just use a dummy signature this is useful for fee estimation
     * @return
     */
   def sign(
-      signers: Seq[Sign],
-      output: TransactionOutput,
-      unsignedTx: Transaction,
-      inputIndex: UInt32,
-      hashType: HashType,
+      spendingInfo: UTXOSpendingInfo,
```

Comment:
So one concern I do have here is that it is possible that the `output` in `UTXOSpendingInfo` and the `output` in `TxSigComponent` could possible be different.

In general, it seems like `TxSigComponent` = `UTXOSpendingInfo` + partially/fully signed transaction that spends that `UTXOSpendingInfo`. Does that sound right? 

I'm not really sure how to handle them interacting with each other, and I don't really have concrete suggestions for making this better, but it does seem like that is a possible invalid state that could be represented currently -- a different output in `TxSigComponent` and `UTXOSpendingInfo`

Reply:
Good point, the only new thing `TxSigComponent` brought to the table was the `unsignedTx`, so I just replaced it with this and derived everything from the two.